static/js/vue-cdr-access/tests/unit/browseSearch.spec.js

```diff
@@ -1,88 +1,95 @@
-const localVue = createLocalVue();
-localVue.use(VueRouter);
-const router = new VueRouter({
-    routes: [
-        {
-            path: '/record/uuid1234',
-            name: 'displayRecords',
-            query: ''
-        }
-    ]
-});
+let wrapper, router;
 
 describe('browseSearch.vue', () => {
-    beforeEach(() => {
+    beforeEach(async () => {
+        router = createRouter({
+            history: createWebHistory(process.env.BASE_URL),
+            routes: [
+                {
+                    path: '/record/:uuid',
+                    name: 'displayRecords',
+                    component: displayWrapper
+                }
+            ]
+        });
         wrapper = shallowMount(browseSearch, {
-            localVue,
-            router,
-            propsData: {
+            global: {
+                plugins: [router]
+            },
+            props: {
                 objectType: 'Folder'
             }
         });
+
+        await router.push('/record/1234');
+
+        wrapper.vm.$router.currentRoute.value.query.anywhere = '';
     });
 
-    it("updates the url when search results change", () => {
+    afterEach(() => router = null);
+
+    it("updates the url when search results change", async() => {
         wrapper.find('input').setValue(query);
         let btn = wrapper.find('button');
-        btn.trigger('click');
-
-        expect(wrapper.vm.$router.currentRoute.query.anywhere).toEqual(encodeURIComponent(query));
+        await btn.trigger('click');
+        await flushPromises();
+        expect(wrapper.vm.$router.currentRoute.value.query.anywhere).toEqual(encodeURIComponent(query));
     });
 
-    it("clears search results", () => {
+    it("clears search results", async() => {
         wrapper.find('input').setValue(query);
         let btn = wrapper.find('button');
-        btn.trigger('click');
-
-        expect(wrapper.vm.$router.currentRoute.query.anywhere).toEqual(encodeURIComponent(query));
+        await btn.trigger('click');
+        await flushPromises();
+        expect(wrapper.vm.$router.currentRoute.value.query.anywhere).toEqual(encodeURIComponent(query));
 
-        let clearLink = wrapper.find('a');
-        clearLink.trigger('click');
-
-        expect(wrapper.vm.$router.currentRoute.query.anywhere).toEqual(encodeURIComponent(''));
+        let clearLink = wrapper.find('a.clear-results');
+        await clearLink.trigger('click');
+        await flushPromises();
+        expect(wrapper.vm.$router.currentRoute.value.query.anywhere).toEqual(encodeURIComponent(''));
     });
 
     it("sets placeholder text from the object type", () => {
         expect(wrapper.find('input').attributes('placeholder')).toBe('Search within this folder');
     });
 
     it("sets default placeholder text if no object type is given", () => {
-        const localVue = createLocalVue();
         const $route = {
-            path: '/record/1234',
+            path: '/record/73bc003c-9603-4cd9-8a65-93a22520ef6a',
```

Comment:
why did this one need to be switched to a UUID and the next test didn't need that change?

Reply:
Changed back. I think I was getting a warning about that uuid.